state/protocol/badger/snapshot.go

```diff
@@ -457,22 +457,22 @@ func (q *EpochQuery) Previous() protocol.Epoch {
 	}
 	if firstHeightKnown && finalHeightKnown {
 		// typical case - we usually know both boundaries for a past epoch
-		return inmem.NewEpochWithStartAndEndBoundaries(setup, commit, firstHeight, finalHeight)
+		return inmem.NewEpochWithStartAndEndBoundaries(setup, entry.PreviousEpoch.EpochExtensions, commit, firstHeight, finalHeight)
```

Comment:
how about moving extension to a variable, it would be easier to detect an inconsistency since for `Previous()` we only need to use previous extension, same for `Current()`